packages/@contentful--app-scripts/lib/get-app-info.js

```diff
@@ -0,0 +1,28 @@
+const { getDefinitionById, selectDefinition } = require('./definition-api');
+const { selectOrganization, getOrganizationById } = require('./organization-api');
+const { createClient } = require('contentful-management');
+const { getManagementToken } = require('./get-management-token');
+
+const getAppInfo = async (options) => {
+  const accessToken = options.token || (await getManagementToken());
+
+  const client = createClient({ accessToken });
+
+  const organization = options.organizationId
+    ? await getOrganizationById(client, options.organizationId)
+    : await selectOrganization(client);
```

Comment:
Refactoring idea (not really a big deal right now, but worth giving it some thought): what about having a `get$ENTITYId` function which takes care of asking things if needed? Something like

```ts
const getOrganizationId = (organizationId?: string) : string => {
  if (organizationId) {
    return organizationId
  }
  
  // ask all the questions and return the appropriate string
}
```

Then the block above would read:

```ts
const organizationId = await getOrganizationId(options.organizationId);
const organization = getOrganizationById(client, organizationId);
```

This way you wouldn't need to implement two methods for each end every entity. Also, the fetching logic in `getOrganizationId` seems quite general, so we can possibly make that just a single utility function and get rid of a lot of code

Reply:
I'm not sure if I understood the approach 100%, but I'm not sure if it would bring more value in that case.
When I understand you correctly, `getOrganizationId` will return the Id in both cases and then fetch it with `getOrganizationById `, to have it unified.

But `selectOrganization` is already returning the organization object what we need. 
So implementing a `getOrganizationId` for both cases and then fetching the organisation in there for all cases would lead to an unnecessary api call when selecting a company and then fetching it again with the id.

Also `getDefinitionById` and `getOrganizationById` look quite similiar, I agree. However there are some slight differences like an additional call in `getDefinitionById` which would make it a bit hard to abstract and might be a bit too much 😬